### PR TITLE
Split quickstarts docs

### PR DESCRIPTION
**Pull request by rsynek**
Removes the quickstarts page (just in Antora docs) and adds individual quickstart docs to the left menu.

quickstarts

### Checklist
- [ ] Documentation updated if applicable.
- [ ] Upgrade recipe provided if applicable.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

We do "simple" maven builds, they are just basically maven commands, but just because we have multiple repositories related between them and one change could affect several of those projects by multiple pull requests, we use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.

[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is not only a github-action tool but a CLI one, so in case you posted multiple pull requests related with this change you can easily reproduce the same build by executing it locally. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* for a <b>pull request build</b> please add comment: <b>Jenkins retest this</b>
* for a <b>specific pull request build</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] tests</b>
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
* for a <b>compile downstream build</b> please add comment: <b>Jenkins run cdb</b>
* for a <b>full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>
* for an <b>upstream build</b> please add comment: <b>Jenkins run upstream</b>
* for a <b>Quarkus LTS check</b> please add comment: <b>Jenkins run LTS</b>
* for a <b>specific Quarkus LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] LTS</b>
* for a <b>Native check</b> please add comment: <b>Jenkins run native</b>
* for a <b>specific Native LTS check</b> please add comment: <b>Jenkins (re)run [optaplanner|kogito-apps|kogito-examples|optaplanner-quickstarts|optaweb-employee-rostering|optaweb-vehicle-routing] native</b>
</details>

### CODE REVIEW

File: optaplanner-docs/src/modules/ROOT/pages/.index.adoc [hunk 1]
```diff
@@ -19,7 +19,7 @@ image::shared/optaPlannerLogo.png[align="center"]
 endif::[]
 
 include::planner-introduction/planner-introduction.adoc[leveloffset=+1]
-include::quickstart/quickstart.adoc[leveloffset=+1]
+include::quickstart/.quickstart.adoc[leveloffset=+1]
```
Comment (rsynek) [inline optaplanner-docs/src/modules/ROOT/pages/.index.adoc:22]: Hidden from Antora, visible to the asciidoc maven plugin

File: optaplanner-docs/src/modules/ROOT/pages/quickstart/overview/overview-quickstarts.adoc [hunk 2]
```diff
@@ -1,5 +1,7 @@
 [[quickStartOverview]]
 = Overview
+// Redirect to this page from .../docs/optaplanner/latest/quickstart/quickstart.html.
+:page-aliases: quickstart/quickstart.adoc
```
Comment (rsynek) [inline optaplanner-docs/src/modules/ROOT/pages/quickstart/overview/overview-quickstarts.adoc:4]: Redirect the previous link.

**Review by yurloc — APPROVED** [review]
Great improvement! This makes it possible to share URLs of individual quickstarts.

**Comment by rsynek** [comment]
> Great improvement! This makes it possible to share URLs of individual quickstarts.

Actually, that is possible even now, but the "summarizing" `quickstart.adoc` (required for single-HTML and PDF), effectively hides them as it's the only entry in the menu.

**Comment by rsynek** [comment]
Btw., we should do the same about "Use cases and examples".

**Pull request merged**